components/TemplateCard/index.js

```diff
@@ -80,11 +107,21 @@ export default function TemplateCard({ templateData }) {
       <span>
         {templateData.days} {templateData.days > 1 ? "Days" : "Day"}
       </span>
-      <IconContainer>
-        <Delete
-          color="crimson"
-          onClick={() => deleteTemplate(templateData._id)}
-        />
+      <IconContainer 
+      style={{backgroundColor: isDelete ? 'var(--sand)' : 'var(--cancel-red)'}}
+      ref={deleteRef}
+      >
+        {
+          !isDelete ? (
+            <Delete
+            onClick={() => setIsDelete(prevState => !prevState)}
+            />
+            ) : (
+              <Checkmark 
+              onClick={() => handleDeleteClick(templateData._id)}
+              />
+            )
+        }
       </IconContainer>
```

Comment:
I think the IconContainer should be a button as this will handle the delete functionality. And the onClick handler should be set on the button and not the icon itself. Then the user also has a larger area where he can click to delete the template. :)

Reply:
Yes, that's absolutely correct. I'm on it.